server/models/db/timeslot.js

```diff
@@ -0,0 +1,24 @@
+const _ = require('lodash');
+
+const { db } = require('../../lib/connection');
+const Tables = require('../../constants/tables.json');
+//const TimeslotStatus = require('../../constants/timeslot-status.json');
+
+// This method returns the timeslot records stored in the timeslot table 
+//Note the parameters, as this method, if given specific start time and specific end time to return, wil only return timeslots within the start and end time parameters 
+//If one or both ofthe parameters is null or not vlaid, the method will return all timeslots in the timeslot table  
+async function getPossibleTimeslots(specificStartTime, specificEndTime) {
+    if (_.isNil(specificStartTime) || _.isNil(specificEndTime)) {
+        const timeslotDetails = await db.any(`select * from ${Tables.timeslot};`);
+        return timeslotDetails;
+    }
+    else {
+        const timeslotDetails = await db.any(`select * from ${Tables.timeslot} where start_time >='${specificStartTime}' and end_time <= '${specificEndTime}';`);
```

Comment:
Hi Peter, 
Notice the duplication of the code in line 12 and line 16: `select * from ${Tables.timeslot}`. A possible way to reduce the repetition is to store this string as a constant or combine the two database queries. For example, we could combine these two if statements to have different values for the queryCondition and use that in the latter half of the query statement as shown below (ROUGH pseudocode):
`var queryCondition = ""
if (!_.isNil(specificStartTime)) queryCondition +- ....
if (!_.isNil(specificEndTime)) queryCondition +- ....
await db.any(select * from.... ${queryCondition ? "where ${queryCondition}; : ";"})`

Reply:
This has been implemented and tested with 100% coverage!